### PR TITLE
Fix #26435: Design System Dojo Dropdown 

### LINKED ISSUE #26435: Design System: Fix Dropdown Search in Dojo
### Problem Statement
The current dropdown is broken when user search for options.

### Steps to Reproduce
1. Go to System > Configuration > Publishing Environments (Tab)
2. Click on Add Environment
3. In the "Who Can Send to this Environment" select, type a letter
4. See the error

image

### Acceptance Criteria
- The dropdown should filter the options when user searches
- We should be able to show 2 lines in the options, for example in the template selector
- Make the highlight string bold

### External Links
N/A

### Assumptions & Initiation Needs
- The search functionality should be consistent across all selects

### Quality Assurance Notes & Workarounds
N/A

### PR DESCRIPTION
**Pull request by zJaaal**
### Summary
🎨♻️🐛

Refactor the dijit menu and combo box styles to use SCSS variables and mixins, and fix the flexbox layout issues for menu items with text only. Move the combo box style from `dotcms.css` to `_menu.scss`.

> _`dijit` menu style_
> _flexbox only when needed_
> _autumn leaves no gaps_

### Walkthrough
* Refactor dijit menu and combo box styles to use SCSS variables and mixins ([link](https://github.com/dotCMS/core/pull/26436/files?diff=unified&w=0#diff-8ced0168f7b85d239121cce95e116f053fbd2c7c5194d010c428d27e5c4b0d39L79-R88), [link](https://github.com/dotCMS/core/pull/26436/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL5046-R5054))

### Screenshots

Screenshot 2023-10-13 at 11 43 52 AM

### CODE REVIEW

**Review by manuelrojas — APPROVED** [review]

File: dotCMS/src/main/webapp/html/css/dijit-dotcms/dotcms.css [hunk 1]
```diff
@@ -5043,13 +5043,15 @@
   margin-top: 0.5rem;
 }
 .dijitComboBoxMenuPopup .dijitMenuItem {
+  padding: 0 0.75rem;
+  color: #14151a;
```
Comment (rjvelazco) [inline dotCMS/src/main/webapp/html/css/dijit-dotcms/dotcms.css:5047]: Is there a variable for this color?

**Review by fmontes — APPROVED** [review]

File: dotCMS/src/main/webapp/html/css/dijit-dotcms/dotcms.css [hunk 2]
```diff
@@ -5043,13 +5043,15 @@
   margin-top: 0.5rem;
 }
 .dijitComboBoxMenuPopup .dijitMenuItem {
+  padding: 0 0.75rem;
```
Comment (rjvelazco) [inline dotCMS/src/main/webapp/html/css/dijit-dotcms/dotcms.css:5046]: Is there any `$space-x` variable that matches this value?

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/webapp/html/css/dijit-dotcms/dotcms.css (hunk 1 above)
Comment (zJaaal) [reply dotCMS/src/main/webapp/html/css/dijit-dotcms/dotcms.css:5047]: This file is the build, it does not have variables, it just generates the css using the homologate .scss file

File: dotCMS/src/main/webapp/html/css/dijit-dotcms/dotcms.css (hunk 2 above)
Comment (zJaaal) [reply dotCMS/src/main/webapp/html/css/dijit-dotcms/dotcms.css:5046]: same

**Review by rjvelazco — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**